src/mlpack/methods/ann/rbm/spike_slab_rbm_policy_impl.hpp

```diff
@@ -0,0 +1,332 @@
+
+#ifndef MLPACK_METHODS_ANN_RBM_SPIKE_SLAB_RBM_POLICY_IMPL_HPP
+#define MLPACK_METHODS_ANN_RBM_SPIKE_SLAB_RBM_POLICY_IMPL_HPP
+
+#include "spike_slab_rbm_policy.hpp"
+
+namespace mlpack {
+namespace ann {
+template<typename DataType>
+SpikeSlabRBMPolicy<DataType>
+      ::SpikeSlabRBMPolicy(const size_t visibleSize,
+      const size_t hiddenSize,
+      const size_t poolSize,
+      const ElemType slabPenalty,
+      ElemType radius):
+      visibleSize(visibleSize),
+      hiddenSize(hiddenSize),
+      poolSize(poolSize),
+      slabPenalty(slabPenalty),
+      radius(2 * radius)
+{
+  parameter.set_size(visibleSize * hiddenSize * poolSize +
+      visibleSize + hiddenSize);
+
+  visibleMean.set_size(visibleSize, 1);
+  spikeMean.set_size(hiddenSize, 1);
+  spikeSamples.set_size(hiddenSize, 1);
+  slabMean.set_size(poolSize, hiddenSize);
+};
+
+// Reset function
+template<typename DataType>
+void SpikeSlabRBMPolicy<DataType>::Reset()
+{
+  // Weight shape D * K * N
+  weight = arma::Cube<ElemType>(parameter.memptr(),
+      visibleSize, poolSize, hiddenSize,
+      false, false);
+
+  // spike bias shape N * 1
+  spikeBias = DataType(parameter.memptr() + weight.n_elem, hiddenSize, 1,
+      false, false);
+
+  // visible penalty 1 * 1 => D * D(when used)
+  visiblePenalty = DataType(parameter.memptr() + weight.n_elem +
+      spikeBias.n_elem, 1, 1, false, false);
```

Comment:
`visibleSize` indicated the number of neurons in the visible layer.
Though the `visiblePenalty` we are using is 1 * 1 scalar. This has to be actually D*D diagonal matrix but we reduced it to 1*1 scalar since for better performance we end using scalar * Identity matrix for `visiblePenalty`